server/middleware/validationMiddleware.ts

```diff
@@ -36,8 +36,8 @@ const zObjectStrict = <T = object>(shape: T) => z.object({ _csrf: z.string().opt
  * more info regarding this issue and workaround on: https://github.com/colinhacks/zod/issues/479#issuecomment-2067278879
  */
 const zodAlwaysRefine = <T extends z.ZodTypeAny>(zodType: T) =>
-  z.any().transform((val, ctx) => {
-    const res = zodType.safeParse(val)
+  z.any().transform(async (val, ctx) => {
```

Comment:
Why does this need to be async? The one in CSIP isn't and just uses `safeParse`

Reply:
in the schema i'm transforming prisonNumber with an api call to prisonerSearch to validate that it's a legit. hence async is needed.

Reply:
Right, but we have api calls in some schemas in CSIP and dont need it :)

Reply:
with CSIP schema we made api calls to get reference data map, then return the zod object, so we only need to `await` to get the zod object. the actual validation itself is all synchronous.

but here we're making the api call when validating the content, hence we need async parse.